### ISSUE TITLE
type alias failing type check after 2.3.1 update

### ISSUE THREAD

**Issue by mlakmal** [Microsoft/TypeScript#15438]
**TypeScript Version:**  2.3.1

```ts
type ResponseTypes = "" | "json" | "text";

interface Test { 
    responseType: string | ResponseTypes;
}

let t: Test = {
    responseType: "json"
};

let tt: ResponseTypes;

tt = t.responseType;
```

**Expected behavior:** should not give error on type checking for above code, but it gives error stating "Type 'string' is not assignable to type 'ResponseTypes'.". this was working fine with 2.2 and start breaking with 2.3.1

**Actual behavior:** should allow proper type checking when using type aliases

**Comment by mhegazy** [comment]
because there is a `string` in the list of types, it is treated as "string".  `string` is a super type of any literal string type. so `string  | "json" | "text"` is just `string`.

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
